include/itkMorphologicalContourInterpolator.hxx

```diff
@@ -0,0 +1,193 @@
+            {
+                if (it->second.GetIndex(a) -1 <= region.GetIndex(a))
+                    it->second.SetIndex(a, it->second.GetIndex(a) - 1);
+                if (it->second.GetIndex(a) + it->second.GetSize(a) + 1 
+                    <= region.GetIndex(a) + region.GetSize(a))
+                    it->second.SetSize(a, it->second.GetSize(a) + 1);
+            }
+        }
+    }
+
+    template< class TImage>
+    void MorphologicalContourInterpolator<TImage>::
+        InterpolateAlong(int axis, typename TImage::Pointer out)
+    {
+
+    }
+
+    template< class TImage>
+    void MorphologicalContourInterpolator<TImage>::GenerateData()
+    {
+        typename TImage::ConstPointer input = this->GetInput();
+        typename TImage::Pointer output = this->GetOutput();
+        this->AllocateOutputs();
+
+        if (m_Axis == -1)
+        {
+            this->DetermineSliceOrientations();
+
+            OrientationType aggregate = OrientationType();
+            if (this->m_Label == 0)
+            {
+                for (OrientationsType::iterator it = m_Orientations.begin(); it != m_Orientations.end(); ++it)
+                    for (int a = 0; a < TImage::ImageDimension; a++)
+                        aggregate[a] = aggregate[a] || it->second[a]; //any label needs interpolation along this axis
+            }
+            else
+                aggregate = m_Orientations[m_Label]; //we only care about this label
+            
+            std::vector<TImage::Pointer> perAxisInterpolates;
+            for (int a = 0; a < TImage::ImageDimension; a++)
+                if (aggregate[a])
+                {
+                    TImage::Pointer imageA = TImage::New();
+                    imageA->CopyInformation(output);
+                    imageA->Allocate();
+                    InterpolateAlong(a, imageA);
+                    perAxisInterpolates.push_back(imageA);
+                }
+            
+            if (perAxisInterpolates.size() == 1)
+            {
+                output = perAxisInterpolates[0];
+                return;
+            }
+            //else
+            std::vector<ImageRegionConstIterator<TImage> > iterators;
+            
+            for (int i = 0; i < perAxisInterpolates.size(); i++)
+            {
+                ImageRegionConstIterator<TImage> it(perAxisInterpolates[i], output->GetRequestedRegion());
+                iterators.push_back(it);
+            }
+
+            std::vector<TImage::PixelType> values;
+            values.reserve(perAxisInterpolates.size());
+
+            ImageRegionIterator<TImage> it(output, output->GetRequestedRegion());
+            while (!it.IsAtEnd())
+            {
+                values.clear();
+                for (int i = 0; i < perAxisInterpolates.size(); i++)
+                {
+                    TImage::PixelType val = iterators[i].Value();
+                    if (val != 0)
+                        values.push_back(val);
+                }
+
```

Comment:
Please summarize the logic used to determine the interpolated value when interpolating over multiple axes in the class Doxygen comment